sentence_transformers/util.py

```diff
@@ -1304,22 +1304,25 @@ def get_device_name() -> Literal["mps", "cuda", "npu", "hpu", "cpu"]:
     return "cpu"
 
 
-def is_accelerate_available() -> bool:
+def check_package_availability(package_name: str, owner: str) -> bool:
     """
-    Returns True if the accelerate library is available.
+    Checks if a package is available from the correct owner.
     """
-    return importlib.util.find_spec("accelerate") is not None
+    spec = importlib.util.find_spec(package_name)
+    if spec and owner in spec.origin:
+        return True
+    return False
```

Comment:
Hello!

Apologies for the delay, I've been recovering from a surgery this last month.
I'd like to improve the `is_..._available` functions to avoid the issue with a `datasets` folder, but I don't believe this works correctly:

```python
# !pip install datasets
>>> import importlib
>>> spec = importlib.util.find_spec("datasets")
>>> spec
ModuleSpec(name='datasets', loader=<_frozen_importlib_external.SourceFileLoader object at 0x7ae5f3430e20>, origin='/usr/local/lib/python3.10/dist-packages/datasets/__init__.py', submodule_search_locations=['/usr/local/lib/python3.10/dist-packages/datasets'])
>>> spec.origin
'/usr/local/lib/python3.10/dist-packages/datasets/__init__.py'
>>> "huggingface" in spec.origin
False
```

Perhaps we can use this:
```suggestion
    try:
        metadata = importlib.metadata.metadata(package_name)
        return metadata["Name"] == package_name and owner in metadata["Home-page"]
    except PackageNotFoundError:
        return False
```

This does work better for me:
```python
>>> from importlib.metadata import metadata
>>> meta = metadata("datasets")
>>> meta["Name"]
'datasets'
>>> meta["Home-page"]
'https://github.com/huggingface/datasets'
>>> meta = metadata("tests") # Note: the tests folder exists locally for me
```

What do you think?

- Tom Aarsen